cli/azd/internal/repository/app_init.go

```diff
@@ -535,6 +537,24 @@ func (i *Initializer) prjConfigFromDetect(
 			}
 		}
 
+		if prj.Language == appdetect.Java && prj.Docker != nil && prj.Docker.Path != "" {
+			hookConfig := ext.HookConfig{
```

Comment:
Per our discussion, this is less efficient

Reply:
In the case of multi-module project

Reply:
Updated to use pack build multi-module in another PR. 
Also will create another work item to track how to fine-grained detect other image builds, such as maven plugin.